packages/react-dom/src/__tests__/ReactDOMTextComponent-test.js

```diff
@@ -197,6 +197,16 @@ describe('ReactDOMTextComponent', () => {
     expect(el.textContent).toBe('');
   });
 
+  it('can reconcile text from pre-rendered markup using dangerouslySetInnerHTML and an object with toString', () => {
```

Comment:
Can we add some tests to `ReactDOMServerIntegration*` suite instead? They will test all possible combinations and verify those match between client-only/server-only/hydration scenarios.